source/s3_client.c

```diff
@@ -48,6 +48,31 @@ PyObject *aws_py_s3_get_recommended_throughput_target_gbps(PyObject *self, PyObj
     return PyFloat_FromDouble(platform_info->max_throughput_gbps);
 }
 
+PyObject *aws_py_s3_get_optimized_platforms(PyObject *self, PyObject *args) {
+    (void)self;
+    (void)args;
+
+    struct aws_array_list platform_list = aws_s3_get_platforms_with_recommended_config();
+
+    size_t list_length = aws_array_list_length(&platform_list);
+
+    PyObject *py_list = PyList_New(list_length);
+    AWS_FATAL_ASSERT(py_list && "platform list allocation failed");
+
+    for (size_t i = 0; i < list_length; ++i) {
+        struct aws_byte_cursor cursor;
+        if (aws_array_list_get_at(&platform_list, &cursor, i) == AWS_OP_SUCCESS) {
+            PyObject *platform_str = PyUnicode_FromStringAndSize((char *)cursor.ptr, cursor.len);
+            AWS_FATAL_ASSERT(platform_str && "platform str allocation failed");
+            PyList_SetItem(py_list, i, platform_str); /* Steals a Reference */
```

Comment:
You must actually handle errors from Py functions. They can fail during interpreter shutdown.

Reply:
@graebm Which function are you talking about? I looked at https://github.com/awslabs/aws-crt-python/blob/main/source/websocket.c#L205 for fatal asserts and noticed that we don't handle the return value of PyList_SetItem in most places.